app/controllers/signatures_controller.rb

```diff
@@ -1,5 +1,6 @@
 class SignaturesController < ApplicationController
   before_filter :retrieve_petition, :only => [:new, :create, :thank_you, :signed]
+  before_filter :retrieve_signature, :only => [:signed]
```

Comment:
You could probably use this new ```before_filter``` in other actions in the controller as well (```verify``` and ```retrieve_signature``` at least)